### ISSUE TITLE
check_http wrong HTTP code

### ISSUE THREAD

**Issue by Hackedia** [nagios-plugins/nagios-plugins#371]
Hi guys !

I want to test the access of my website http://anothersite.com/login with port 80 open and port 443 CLOSED :

check_http -H anothersite.com -p 80 -u /login
=> HTTP OK: HTTP/1.1 301 Moved Permanently - 468 bytes in 0.001 second response time
check_http -H anothersite.com -S -p 443 -u /login
=> HTTP OK: HTTP/1.1 200 OK - 2512 bytes in 0.013 second response time

So i dont understand how check_http access my website on port 443 and how he meet ERROR 301 on port 80,
Does anyone understand the behavior of the script here ? (I tryed every syntaxe here without getting the right return i was guessing),

Thanks for your support and have a nice day !

**Comment by hedenface** [comment]
Is the web server redirecting non-SSL traffic to SSL? That's the first thought that comes to mind.

It's going to be really hard to debug something like this without knowing the actual website you're testing.

**Comment by Hackedia** [comment]
Yeah i understand, in this case i would have read the script but its a too advanced C script for me ^_^,
I don't know how exactly the script ask the website but when i connect to the site on HTTPS, i got a 404 ERROR message, how do the script to get a 200 OK error message instead of 404 !

**Comment by hedenface** [comment]
Can you show me the exact commands and the exact website you used for your test?

I don't think I can come up with anything other than guesses if I don't have that information.

**Comment by Hackedia** [comment]
Ok, thanks for your help hedenface !

Here are the commands and the results :
check_http -H e-learning.ch-larochelle.fr -p 80 -u /login -N
=> HTTP OK: HTTP/1.1 301 Moved Permanently - 468 bytes in 0.001 second response time
check_http -H e-learning.ch-larochelle.fr -S -p 443 -u /login -N
=> HTTP OK: HTTP/1.1 200 OK - 2512 bytes in 0.013 second response time

**Comment by hedenface** [comment]
Okay, well when I go there in my browser, I get a 303. According to Mozilla:

> `303 See Other` redirect status response code indicates that the redirects don't link to the newly uploaded resources but to another page, like a confirmation page or an upload progress page

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303

As far as the 443 thing, I don't get a 404 at all when I visit that page. Is your browser configured properly? Can you try it on another browser or computer?

Or did you confuse the "Security Exception" with 404? If you're using Firefox, you'll need to add an exception in order to view that page.

In any case, it appears that your `check_http` is working appropriately.

**Comment by Hackedia** [comment]
Hi hedenface, 
And by the way thanks for your help :),

To answer your questions :

> As far as the 443 thing, I don't get a 404 at all when I visit that page.
You succeed connecting to the website on https (https://e-learning.ch-larochelle.fr) ? I get an error 404 again.
>Can you try it on another browser or computer?
Its the same with chrome and firefox,
> Or did you confuse the "Security Exception" with 404?
I got this page, accetped the exception and after get the 404 error ^_^,

Or maybe im not doing this right, but for me connecting to the website though https is the same as asking the port 443 of the apache server, no ?
And the sysadmin tel me that the website is not supposed to work though 443 https !

I don't get it anyway !
